ta4j-core/src/main/java/org/ta4j/core/indicators/RWIHighIndicator.java

```diff
@@ -57,25 +57,28 @@ public RandomWalkIndexHighIndicator(TimeSeries series, int timeFrame) {
 
     @Override
     protected Decimal calculate(int index) {
-        int lastIndex = Math.max(0, index - timeFrame+1);
-        int n = 2;
-        AverageTrueRangeIndicator averageTrueRange = new AverageTrueRangeIndicator(getTimeSeries(), n);
-        Decimal highestRWI = maxPrice.getValue(index).minus(minPrice.getValue(Math.max(0, index-1)))
-                .dividedBy(averageTrueRange.getValue(index).multipliedBy(Decimal.valueOf(Math.sqrt(n))));
-
-        for(int i = index-2; i >= lastIndex; i--) {
-            n++;
-            averageTrueRange = new AverageTrueRangeIndicator(getTimeSeries(), n);
-            Decimal currentRWI = maxPrice.getValue(index).minus(minPrice.getValue(i))
-                    .dividedBy(averageTrueRange.getValue(index).multipliedBy(Decimal.valueOf(Math.sqrt(n))));
-            if(currentRWI.isGreaterThan(highestRWI)){
-                highestRWI = currentRWI;
-            }
-
+        Decimal highestRWI = Decimal.NaN;
+        
+        for(int n = 2; n <= this.timeFrame; n++) {
+           Decimal currentRWI = calcRWIHighValue(index, index-n, n);
+           if(currentRWI.isGreaterThan(highestRWI)){
+               highestRWI = currentRWI;
+           }   
         }
         return highestRWI;
     }
     
+    /**
+     * @param t = current index
+     * @param t_n = current index - n
+     * @param n = n starting at 2 increments until n = time frame
+     */
+    private Decimal calcRWIHighValue(int t, int t_n, int n){
```

Comment:
no need for method-parameter "t_n".

Say "t-n" directly within the method:
```
private Decimal calcRWIHighValue(int t, int n){
..
return maxPrice.getValue(t).minus(minPrice.getValue(t - n))
}
```

and rename param "t" to "index":

```
private Decimal calcRWIHighValue(int index, int n){
AverageTrueRangeIndicator atr = new AverageTrueRangeIndicator(getTimeSeries(), n);
      return maxPrice.getValue(index).minus(minPrice.getValue(index - n))
                  .dividedBy(atr.getValue(t).multipliedBy(Decimal.valueOf(Math.sqrt(n))));
   }
```